rust/signed_doc/src/metadata/doc_type.rs

```diff
@@ -83,6 +118,25 @@ impl TryFrom<Vec<UuidV4>> for DocType {
     }
 }
 
+impl TryFrom<Vec<String>> for DocType {
+    type Error = DocTypeError;
+
+    fn try_from(value: Vec<String>) -> Result<Self, Self::Error> {
+        if value.is_empty() {
+            return Err(DocTypeError::Empty);
+        }
+        let converted = value
+            .into_iter()
+            .map(|s| {
+                let parsed = Uuid::from_str(&s).map_err(|_| DocTypeError::StringConversion(s))?;
```

Comment:
I believe it make sense to add a `FromStr` trait implementation for the `UuidV4` and `UuidV7` type, so we would be able to make a strict conversion from string to our UUID types